File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.cpp
#include <iostream>
#include <string>
#include <stack>
using namespace std;

int solution(string s)
{
    int answer = 0;

    if(s.size() % 2 == 1) return answer;
    
    stack<char> st;
    for(int i = 0; i<s.size(); i++){
        if(st.empty() || st.top() != s[i]){
            st.push(s[i]);
        }
        else{
            st.pop();
        }
    }

    if(st.empty()) answer = 1;
    return answer;
}

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.*;

class Solution
{
    public int solution(String s)
    {
        int answer = 0;

        Stack<Character> st = new Stack<>();
        for(char c : s.toCharArray()) {
            if(!st.isEmpty()) {
                if(st.peek() == c) st.pop();
                else st.push(c);
            }
            else {
                st.push(c);
            }
        }
        
        if(st.isEmpty()) answer = 1;

        return answer;
    }
}